lib/internal/bootstrap/pre_execution.js

```diff
@@ -369,6 +370,11 @@ function initializeCJSLoader() {
 }
 
 function initializeESMLoader() {
+  // proxify the global process to keep in sync with ESM
+  const mod = NativeModule.map.get('process');
+  mod.compileForPublicLoader(true);
+  globalThis.process = mod.exports;
```

Comment:
If this is done when ESM is not even used doesn't this just slow down access to `globalThis.process` unconditionally?

Reply:
Yup, but that's how to keep ESM names synced